pkg/controller/infrastructure/actuator_reconcile.go

```diff
@@ -34,18 +36,56 @@ func (a *actuator) reconcile(ctx context.Context, infrastructure *extensionsv1al
 		return err
 	}
 
-	firewallStatus := infrastructureStatus.Firewall
-	if firewallStatus.Succeeded {
-		return nil
-	}
-
 	mclient, err := metalclient.NewClient(ctx, a.client, &infrastructure.Spec.SecretRef)
 	if err != nil {
 		return err
 	}
 
+	clusterID := cluster.Shoot.Status.TechnicalID
+	clusterTag := fmt.Sprintf("%s=%s", metal.ShootAnnotationClusterID, clusterID)
+
+	firewallStatus := infrastructureStatus.Firewall
+
+	if firewallStatus.Succeeded {
+		// verify that the firewall is still there and correctly reconciled
+		machineID := decodeMachineID(firewallStatus.MachineID)
+
+		resp, err := mclient.FirewallFind(&metalgo.FirewallFindRequest{
+			MachineFindRequest: metalgo.MachineFindRequest{
+				ID:                &machineID,
+				AllocationProject: &infrastructureConfig.ProjectID,
```

Comment:
we can add clusterid tag here as well to narrow down the search term. Just thought.

Reply:
Good idea, this also reduces the amount of changes.